server/src/main/java/org/elasticsearch/transport/TransportService.java

```diff
@@ -654,7 +654,7 @@ public String toString() {
             if (ex instanceof TransportException) {
                 te = (TransportException) ex;
             } else {
-                te = new TransportException("failure to send", ex);
+                te = new SendRequestTransportException(getLocalNode(), "failure to send", ex);
```

Comment:
We should supply the target node as well as the action to the exception.

```suggestion
                te = new SendRequestTransportException(connection.getNode(), action, ex);
```